var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -614,6 +623,13 @@ def cmake_args(self):
             options.append("-DGMX_GPU_NB_CLUSTER_SIZE=8")
             options.append("-DGMX_GPU_NB_NUM_CLUSTER_PER_CELL_X=1")
 
+        if "+itt" in self.spec:
+            options.append("-DGMX_USE_ITT=on")
+            options.append(
+                "-DITTNOTIFY_INCLUDE_DIR=%s"
+                % join_path(self.spec["intel-oneapi-vtune"].package.headers)
```

Comment:
Is this a directory?

Reply:
Yes, see https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/oneapi.py#L270. This is the first time we've used one of these oneapi interfaces. Is there a vision for a consistent approach across Spack things? I see a bit of use of `headers.directories`.

Thoughts @rscohn2 ?

Reply:
Ugh, I see:
- The callback is expected to return an `HeaderList` (usually via a call to `find_headers`)
- But since Python is Python, `oneapi` does it differently and we get no warnings 

:sob: @rscohn2 yeah, we should change `oneapi` to match what the `headers` interface expects as a return value.

Reply:
@mabraham That said, it's not a fault of this PR, so we can merge this and make `oneapi` consistent with the other callbacks in another PR.

Reply:
https://github.com/spack/spack/issues/47791